### PR TITLE
Fix RPC undefined behavior

### PR DESCRIPTION
**Pull request by rex4539**
Closes https://github.com/zcash/zcash/issues/3792

### CODE REVIEW

File: src/arith_uint256.cpp [hunk 1]
```diff
@@ -173,8 +173,8 @@ unsigned int base_uint<BITS>::bits() const
 {
     for (int pos = WIDTH - 1; pos >= 0; pos--) {
         if (pn[pos]) {
-            for (int bits = 31; bits > 0; bits--) {
-                if (pn[pos] & 1 << bits)
+            for (size_t bits = 31; bits > 0; bits--) {
+                if (pn[pos] & 1U << bits)
```
Comment (daira) [inline src/arith_uint256.cpp]: Please write this as `pn[pos] & (1U << bits)`.

Comment (rex4539) [reply src/arith_uint256.cpp]: Fixed.

File: src/arith_uint256.cpp [hunk 2]
```diff
@@ -173,8 +173,8 @@ unsigned int base_uint<BITS>::bits() const
 {
     for (int pos = WIDTH - 1; pos >= 0; pos--) {
         if (pn[pos]) {
-            for (int bits = 31; bits > 0; bits--) {
-                if (pn[pos] & 1 << bits)
+            for (size_t bits = 31; bits > 0; bits--) {
+                if pn[pos] & (1U << bits)
```
Comment (Eirik0) [inline src/arith_uint256.cpp]: Nit: I would prefer this if it were `if (pn[pos] & (1U << bits))`, but this is non-blocking.

Comment (daira) [reply src/arith_uint256.cpp]: Huh, that's not even valid C++ as it is. Will fix.

**Comment by daira** [comment]
Needs another review because I co-authored.

**Comment by ebfull** [comment]
@zkbot r+

**Pull request merged**